packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/common.ts

```diff
@@ -1,6 +1,43 @@
 import * as iam from '@aws-cdk/aws-iam';
 import * as logs from '@aws-cdk/aws-logs';
 
+/**
+ * Possible compression options Kinesis Data Firehose can use to compress data on delivery.
+ */
+export class Compression {
+  /**
+   * gzip
+   */
+  public static readonly GZIP = new Compression('GZIP');
+
+  /**
+   * Hadoop-compatible Snappy
+   */
+  public static readonly HADOOP_SNAPPY = new Compression('HADOOP_SNAPPY');
+
+  /**
+   * Snappy
+   */
+  public static readonly SNAPPY = new Compression('Snappy');
+
+  /**
+   * Uncompressed
+   */
+  public static readonly UNCOMPRESSED = new Compression('UNCOMPRESSED');
+
+  /**
+   * ZIP
+   */
+  public static readonly ZIP = new Compression('ZIP');
+
+  constructor(
+    /**
+     * String value of the Compression.
+     */
+    public readonly value: string,
+  ) { }
```

Comment:
This is how I'd write it:

```suggestion
  /**
   * Creates a new Compression instance.
   *
   * @param value the string value of the compression
   */
  constructor(public readonly value: string) {
  }
```

Reply:
BTW, I like having a static factory method (something like `of(): Compression`), and making the constructor private, more than making the constructor public. 2 reasons:

1. A static factory method is more discoverable (a customer uses this class as `Compression.`, the IDE will show them the static factory method, while they might miss the constructor).
2. You get a little more flexibility in the future (for example, you can introduce module-private subclasses of `Compression` later on, and make that be a backwards-compatible change).